crates/astria-sequencer/src/transaction/unsigned.rs

```diff
@@ -113,8 +113,10 @@ impl ActionHandler for Unsigned {
         state: &S,
         from: &Address,
     ) -> Result<()> {
+        // Nonce should be equal to the number of executed transactions before this tx.
+        // First tx has nonce 0.
         let curr_nonce = state.get_account_nonce(from).await?;
-        ensure!(curr_nonce < self.nonce, "invalid nonce");
+        ensure!(curr_nonce == self.nonce, "invalid nonce");
```

Comment:
```suggestion
        ensure!(curr_nonce == self.nonce, "transaction nonce must match stored account nonce");
```
Right now the reason gets lost:
```rust
fn run() -> eyre::Result<()> {
    let x = 5;
    let y = 6;
    eyre::ensure!(x == y, "invalid bla");
    Ok(())
}

fn main() {
    if let Err(e) = run() {
        println!("{e:?}");
        // invalid bla
        //
        // Location:
        //    src/main.rs:4:5
    }
}
```

Reply:
Yeah, I'm not sure how much this is intended behaivior or not. I know that when you submit an invalid sequence to a Cosmos chain you don't get any useful feedback, just that it's invalid.

Reply:
Yeah, this is more for a user not having to dig into sequencer when this error happens. `invalid nonce` reads a bit like `error occured`. So like - yeah, great. But what exactly was the problem now?